kyuubi-hive-jdbc/pom.xml

```diff
@@ -86,6 +86,11 @@
             <artifactId>hive-service-rpc</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>org.apache.hadoop</groupId>
+            <artifactId>hadoop-common</artifactId>
```

Comment:
we spent significant efforts to make the Kyuubi JDBC driver become a Hadoop-free component, so that the consumers won't pull bunches of Hadoop transitive deps.

Reply:
to keep behavior with Hive JDBC, e.g. support Hadoop UGI authentication, we use reflection to check Hadoop classes available and for sequential method invocations

Reply:
remove hadoop dependency, and use reflection to check Hadoop classes. Help review the PR, thanks.